### DISCUSSION TITLE
Example of queries

### DISCUSSION THREAD

**Discussion by lezhnev74** [opcodesio/log-viewer#212] [Q&A] [answered]
Hey! Could you share examples of how to use the query for searching strings: `A OR B` or `A AND B` or `A AND NOT B`?
That would be a great documentation page.

**Comment by arukompas** [comment, marked as answer]
hey @lezhnev74 , the search is **RegEx** capable, in the sense that it just puts everything in the search between `/` and `/i`.

So, for example, to search for `A OR B`, you could just write `A|B`.
To search for `A AND B`, you could write `A.*B` (if A comes first), or `B.*A` (if B comes first)
To search for `A AND NOT B`, you could write `A.*(?!B)`

I'm not super savvy with RegEx, but at least now you know how the search works :) should definitely add this somewhere in the docs

**Reply by lezhnev74** [reply]
I see something strange here, it won't let me find values in multiline messages like:
```
{
  "id":"12",
  "type": "Account"
}
```

The query `Account` find the message but `12.*Account` does not. I am thinking about multiline regexp, do you have a test somewhere that shows how multiline search works in this program?

**Reply by lezhnev74** [reply]
haha, well this one worked `12(.|\n)*Account`

**Reply by arukompas** [reply]
Why did I think multi-line worked before 😲 `12.*Account` should work! I'll have a look and maybe push a change to improve it :) 

Thanks for your feedback!
